botorch/acquisition/risk_measures.py

```diff
@@ -47,18 +47,20 @@ class RiskMeasureMCObjective(MCAcquisitionObjective, ABC):
     def __init__(
         self,
         n_w: int,
-        weights: Optional[Tensor] = None,
+        weights: Optional[Union[List[float], Tensor]] = None,
     ) -> None:
         r"""Transform the posterior samples to samples of a risk measure.
 
         Args:
             n_w: The size of the `w_set` to calculate the risk measure over.
-            weights: An optional `m`-dim tensor of weights for scalarizing
+            weights: An optional `m`-dim tensor or list of weights for scalarizing
                 multi-output samples before calculating the risk measure.
         """
         super().__init__()
         self.n_w = n_w
-        self.weights = weights
+        self.register_buffer(
+            "weights", torch.as_tensor(weights) if weights is not None else None
+        )
```

Comment:
I had to check what this does - if `None` this doesn't actually register a buffer at all on the module. I feel this is kind of unclear, so let's do this instead.
```suggestion
        if weights is not None:
            self.register_buffer("weights", torch.as_tensor(weights))
```

Reply:
So, it actually registers it. It just won't display `None` buffers. If you do `self.register_buffer("weights", None)` and call `self.weights`, it will not raise an `AttributeError`. `self.weights` is `None`, and you can find `weights` under `self._buffers`. It just doesn't get included in `state_dict` or in `repr(self)`.

Reply:
hmm, interesting. Ok that sounds reasonable then